modmail/exts/utils/extensions.py

```diff
@@ -0,0 +1,278 @@
+# original source:
+# https://github.com/python-discord/bot/blob/a8869b4d60512b173871c886321b261cbc4acca9/bot/exts/utils/extensions.py  # noqa: E501
+# MIT License 2021 Python Discord
+import functools
+import logging
+import typing as t
+from enum import Enum
+
+from discord import Colour, Embed
+from discord.ext import commands
+from discord.ext.commands import Context, group
+
+from modmail import exts
+from modmail.bot import ModmailBot
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import CogMetadata
+from modmail.utils.extensions import EXTENSIONS, unqualify
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+
+UNLOAD_BLACKLIST = {
+    __name__,
+}
+BASE_PATH_LEN = len(exts.__name__.split("."))
+
+COG_METADATA = CogMetadata(develop=True)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(ModmailBot.load_extension)
+    UNLOAD = functools.partial(ModmailBot.unload_extension)
+    RELOAD = functools.partial(ModmailBot.reload_extension)
+
+
+class Extension(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if argument == "*" or argument == "**":
+            return argument
+
+        argument = argument.lower()
+        extensions = []
+        for ext, _nul in EXTENSIONS:
```

Comment:
This seems to be getting repeated at a few places, can we take this to the top of the file  as `EXTENSION_NAMES` or something like that? Since this is just a for loop, it can be ignored but I had like this change.